src/main/kotlin/com/ampnet/reportserviceeth/service/data/CurrencyFormatting.kt

```diff
@@ -3,9 +3,7 @@ package com.ampnet.reportserviceeth.service.data
 import java.math.BigInteger
 import java.text.DecimalFormat
 
-const val FROM_CENTS_TO_EUROS = 100.0
+const val DECIMALS_PRECISION = 1_000_000_000_000_000_000
 
-fun Long.toEurAmount(): String = DecimalFormat("#,##0.00").format(this / FROM_CENTS_TO_EUROS)
-
-// TODO("this -> Gwei to ether -> format")
-fun BigInteger.toEurAmount(): String = DecimalFormat("#,##0.00").format(this.toLong() / FROM_CENTS_TO_EUROS)
+fun BigInteger.toEther(): String = DecimalFormat("#,##0.00")
+    .format(this / BigInteger(DECIMALS_PRECISION.toString()))
```

Comment:
```suggestion
    .format(this / BigInteger.valueOf(DECIMALS_PRECISION))
```
It doesn't make sense to convert Long to String then to BigInteger.